Add dashed support for legends

mollerjorge opened the pull request:
## What does this implement/fix?

1. We need to add support for showing a dashed line in the legend. The use case is for benchmarks where the `Median` line is dashed, so the legend should follow the same pattern.
2. Fixed Annotations textAnchor property, it shouldn't be `left` `middle` `right` but `start` `middle` `end`
3. Fix text alignment on annotations: Should be centered

Before:
<img width="1155" alt="Screenshot 2024-12-04 at 10 44 27 AM" src="https://github.com/user-attachments/assets/61748de3-67e6-4b75-bd56-698fa77a0675">

After:
<img width="1173" alt="Screenshot 2024-12-04 at 10 44 07 AM" src="https://github.com/user-attachments/assets/5f60b1d2-beba-490a-b3be-eb0f018aa922">

🎩  Tophatting:

1. Annotations text alignment: https://x5mqehaaczjinobf.tunnel.shopifycloud.tech/?path=/story/polaris-viz-charts-linechart--annotations
2. Dashed legend in line chart relational: https://x5mqehaaczjinobf.tunnel.shopifycloud.tech/?path=/story/polaris-viz-charts-linechartrelational--default

### Before merging

- [ ] Check your changes on a variety of [browsers](https://help.shopify.com/en/manual/shopify-admin/supported-browsers) and devices.

- [ ] Update the Changelog's Unreleased section with your changes.

- [ ] Update relevant documentation, tests, and Storybook.

- [ ] Make sure you're exporting any new shared Components, Types and Utilities from the top level index file of the package

mollerjorge commented on packages/polaris-viz/src/components/LineChartRelational/stories/data.tsx:
```diff
@@ -105,6 +105,10 @@ export const DEFAULT_DATA: DataSeries[] = [
       {value: 21, key: '2020-03-14T12:00:00'},
     ],
     color: LIGHT_THEME.seriesColors.limited[5],
+    metadata: {
```
using metadata for this looks like a good fit after checking other like:

```
metadata: {
      relatedIndex: 2,
      areaColor: 'rgba(218, 182, 242, 0.2)',
      legendLabel: '75th - 25th percentile',
    },
```

mollerjorge commented on packages/polaris-viz/src/components/shared/SeriesIcon/SeriesIcon.tsx:
```diff
@@ -1,30 +1,20 @@
-}: Props) {
-  const selectedTheme = useTheme();
-
+export function SeriesIcon({color, lineStyle, shape = 'Bar'}: Props) {
```
Thought of refactoring this a bit, it didn't feel like SeriesIcon should concern about wether if its a comparison or not, I think we should pass in the `color` and `lineStyle` instead and calculate those from the parents

carysmills commented on packages/polaris-viz/CHANGELOG.md:
```diff
@@ -9,6 +9,10 @@ and adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [15.3.4] - 2024-12-03
 
+### Added
```
This should go under "unreleased" not 15.3.4

carysmills reviewed (approved):
Code looks good to me except the changelog
Did not 🎩 as @envex has more context on this chart and would be good for him to have a quick look

mollerjorge commented:
> Code looks good to me except the changelog Did not 🎩 as @envex has more context on this chart and would be good for him to have a quick look

Thanks @carysmills ! I added a link to tophat so its easier now, and added a fix for annotations as well.
